plugins/okta/main.go

```diff
@@ -0,0 +1,461 @@
+	} `json:"transaction"`
+	DebugContext struct {
+		DebugData struct {
+			RequestURI        string `json:"requestUri"`
+			OriginalPrincipal struct {
+				ID          string `json:"id,omitempty"`
+				Type        string `json:"type,omitempty"`
+				AlternateID string `json:"alternateId,omitempty"`
+				DisplayName string `json:"displayName,omitempty"`
+			} `json:"originalPrincipal,omitempty"`
+		} `json:"debugData,omitempty"`
+	} `json:"debugContext,omitempty"`
+	AuthenticationContext struct {
+		AuthenticationStep int    `json:"authenticationStep,omitempty"`
+		ExternalSessionID  string `json:"externalSessionId,omitempty"`
+	} `json:"authenticationContext,omitempty"`
+	SecurityContext struct {
+		AsNumber int    `json:"asNumber,omitempty"`
+		AsOrg    string `json:"asOrg,omitempty"`
+		ISP      string `json:"isp,omitempty"`
+		Domain   string `json:"domain,omitempty"`
+	} `json:"securityContext,omitempty"`
+}
+
+// OktaPlugin represents our plugin
+type OktaPlugin struct {
+	plugins.BasePlugin
+	APIToken     string `json:"api_token" jsonschema:"description=API Token,required"`
+	Organization string `json:"organization" jsonschema:"description=Your Okta organization,required"`
+	lastLogEvent LogEvent
+	lastEventNum uint64
+}
+
+// OktaInstance represents a opened stream based on our Plugin
+type OktaInstance struct {
+	source.BaseInstance
+	client      *http.Client
+	request     *http.Request
+	cancel      context.CancelFunc
+	lastReqTime time.Time
+}
+
+const oktaBaseURL string = "okta.com/api/v1/logs"
+
+// init function is used for referencing our plugin to the Falco plugin framework
+func init() {
+	p := &OktaPlugin{}
+	extractor.Register(p)
+	source.Register(p)
+}
+
+// Info displays information of the plugin to Falco plugin framework
+func (oktaPlugin *OktaPlugin) Info() *plugins.Info {
+	return &plugins.Info{
+		ID:                 7,
+		Name:               "okta",
+		Description:        "Okta Log Events",
+		Contact:            "github.com/falcosecurity/plugins/",
+		Version:            "0.1.0",
+		RequiredAPIVersion: "0.3.0",
+		EventSource:        "okta",
+	}
+}
+
+// InitSchema exports the json schema for parameters
+func (oktaPlugin *OktaPlugin) InitSchema() *sdk.SchemaInfo {
+	reflector := jsonschema.Reflector{
+		RequiredFromJSONSchemaTags: true, // all properties are optional by default
+		AllowAdditionalProperties:  true, // unrecognized properties don't cause a parsing failures
+	}
+	if schema, err := reflector.Reflect(&OktaPlugin{}).MarshalJSON(); err == nil {
+		return &sdk.SchemaInfo{
+			Schema: string(schema),
+		}
+	}
+	return nil
+}
+
+// Init is called by the Falco plugin framework as first entry,
+// we use it for setting default configuration values and mapping
+// values from `init_config` (json format for this plugin)
+func (oktaPlugin *OktaPlugin) Init(config string) error {
+	err := json.Unmarshal([]byte(config), &oktaPlugin)
+	if err != nil {
+		return err
+	}
+	if oktaPlugin == nil {
```

Comment:
These checks are probably useless since you have an init schema (the framework is supposed to pre-validate your config before providing it to Init()). You can make all the fields mandatory by using the `RequiredFromJSONSchemaTags: true` field in the `jsonschema.Reflector` struct you use in InitSchema().